src/main/java/com/gradle/develocity/bamboo/ShortLivedTokenClient.java

```diff
@@ -0,0 +1,64 @@
+package com.gradle.develocity.bamboo;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class ShortLivedTokenClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShortLivedTokenClient.class);
+
+    private static final RequestBody EMPTY_BODY = RequestBody.create(new byte[]{});
+
+    private final OkHttpClient httpClient;
+
+    public ShortLivedTokenClient() {
+        this.httpClient = new OkHttpClient().newBuilder()
+                .connectTimeout(10, TimeUnit.SECONDS)
+                .writeTimeout(10, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .build();
+    }
+
+    public Optional<DevelocityAccessKey> get(String server, DevelocityAccessKey accessKey, String expiryInHours) {
+        String url = normalize(server) + "api/auth/token";
+        if (StringUtils.isNotBlank(expiryInHours)) {
+            url += "?expiresInHours=" + expiryInHours;
+        }
+
+        Request request = new Request.Builder()
```

Comment:
I recommend implementing some form of a retry mechanism. That way, if the Develocity instance is down briefly, the build has a chance of still working.

We found a retry mechanism to be quite helpful in our own dogfooding of short-lived access tokens.

Reply:
Thanks for the proposal, I've added the retry component